R/http.R

```diff
@@ -142,7 +142,12 @@ handleResponse <- function(response, error_call = caller_env()) {
   if (isContentType(response$contentType, "application/json")) {
     # parse json responses
     if (nzchar(response$content)) {
-      json <- jsonlite::fromJSON(response$content, simplifyVector = FALSE)
+      json <- tryCatch({
+        jsonlite::fromJSON(response$content, simplifyVector = FALSE)
+      }, error = function(e) {
+
+        return(response$content)
```

Comment:
When is a response that claims to be `application/json` not JSON? Why is that no longer an error?

Reply:
Maybe it would be preferable to condition this behavior on responses from snowflake hosts only?

Reply:
Could you explain the situation where this occurs?

Reply:
The access token returned by the exchange is just the response body, [rather than the JSON message the spec would lead you to expect.](https://github.com/posit-dev/snowflakeauth/blob/fa679d898f151f766c20f97e0ff5188ed4b99ef3/R/keypair.R#L90-L94)

Reply:
Oh, that's really unfortunate. As this is only for a single call path (to obtain the access token), is it worth having a separate HTTP helper for that?

Another idea: add an argument to the `POST` call indicating that a "raw" response should be returned. That would be passed down to `handleResponse` and this code becomes:

```r
if (rawResponse) {
    if (response$status %in% 200:399) {
      out <- response$content
    } else {
      reportError(response$content)
    }
} else if (isContentType(response$contentType, "application/json)) {
  # existing code
} # ...
```

This basically duplicates the final clause, but for a different reason.

Reply:
The introduction of `rawResponse = TRUE` from `GET_server_settings` appears to have caused https://github.com/rstudio/rsconnect/issues/1166.